tests/falcon_test.py

```diff
@@ -88,3 +119,32 @@ class HelloResource:
         spec.path(resource=hello_resource)
 
         assert spec._paths["/hi"]["x-extension"] == "global metadata"
+
+    def test_unredundant_basepath_resource(self, app, spec_factory, settings):
+        class HelloResource:
+            def on_get(self, req, resp):
+                """A greeting endpoint.
+                ---
+                description: get a greeting
+                responses:
+                    200:
+                        description: said hi
+                """
+                return "dummy"
+
+        expected = {
+            "description": "get a greeting",
+            "responses": {"200": {"description": "said hi"}},
+        }
+        full_path = "/v1/foo/v1"
+        path_expected = "/foo/v1"
```

Comment:
Do these need to be variables? I would just reference them directly in the test.